contracts/src/BeefyClient.sol

```diff
@@ -212,9 +230,14 @@ contract BeefyClient {
         external
     {
         ValidatorSet memory vset;
+        uint16 signatureCount;
         if (commitment.validatorSetID == currentValidatorSet.id) {
+            signatureCount = currentValidatorSetCounters.get(proof.index);
+            currentValidatorSetCounters.set(proof.index, signatureCount + 1);
```

Comment:
I kinda think we should use saturating arithmetic here. Otherwise the submission will fail on overflow. Obviously this is unlikely to happen to act in practice, but we should be paranoid just in case.

```suggestion
            currentValidatorSetCounters.set(proof.index, signatureCount.saturatingAdd(1));
```

Will need to add a `saturatingAdd(uint16 value)` helper to `Math.sol`.

Reply:
Addressed in https://github.com/Snowfork/snowbridge/pull/956/commits/b9b7127027e342adde23c43af16d46dea6d92db2